### ISSUE TITLE
Set filter's get filter model returns "null" for a null value instead of null

### ISSUE THREAD

**Issue by nickytonline** [ceolter/ag-grid-enterprise#109]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/ceolter/ag-grid-enterprise/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

1. Pass an array of values to filterParams for a 'set' filter or let ag-grid deduce the unique values in the grid where the list of unique values will contain a `null`.
2. Whenever `api.getFilterModel()` is called, it returns the selected values of the 'set' filter via `Utils.iterateObject`. Under the hood it is an object and since an object can only use strings as keys, `null` gets converted to `"null"`, which is what causes the bug.

**Expected behavior**

If a `null` value is in the list of unique values for a 'set' filter, it's value returned, when calling `api.getFilterModel()` should return `null` for a null value, not `"null"`.

**Minimal reproduction of the problem with instructions**

Can't add ag-grid-enterprise in a plunker. Our stuff is not publicly available on the web and I can only find CDNs for ag-grid.

**What is the motivation / use case for changing the behavior?**

Basically what I described above.

I think the cleanest way to do this given the current implementation is to return `null` when the `"null"` key is found, otherwise just return the key.

An improvement to the implementation would be to use an ES6 `Map` because the keys of the map can be anything, including `null`. Since ag-grid is using TypeScript, you will be able to transpile an ES6 Map to ES5.

**Please tell us about your environment:**

Ubuntu 16.10
webpack 1.13.1
VS Code 1.9.1

* ag-grid@7.2.2
* ag-grid-enterprise@6.2.0
* ag-grid-react@7.2.0

* **Browser:** [Chrome 54|IE 11 but this probably happens in all browsers]

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
typescript@2.0.0 transpiled to ES5

**Comment by nickytonline** [comment]
Also, it's still an issue in the current version, 8.0.1

**Comment by willhalling** [comment]
Thanks for raising @nickytonline 

This should be fixed in a future release. You can track progress on our pipeline (AG-256):

https://ag-grid.atlassian.net/browse/AG-256

**Comment by colorgap** [comment]
Also i noticed that number filter is broken in V 8.x.  Currently using v8.2.0 equal and not equals filter are not working also if you try lessthanorequalsto 0 or greaterthanandequals to 0. thats broken too.

It used to work in v7.x

**Comment by makinggoodsoftware** [comment]
Hi, This all should be fixed now.
